### ISSUE TITLE
options do not get set in distinct query

### ISSUE THREAD

**Issue by clhuang** [Automattic/mongoose#8907]
**Do you want to request a *feature* or report a *bug*?**
bug
**What is the current behavior?**
readPreference and maxTimeMS options are ineffective on a distinct query

**If the current behavior is a bug, please provide the steps to reproduce.**
```js
mongoose.set('debug', true);
await mongoose.connect(MONGO_URI);
const Model = mongoose.model(
  'model',
  ...
);
await Model.distinct('field').read('secondary');
```

_debug output:_
`Mongoose: models.distinct('field', {})`
query appears in active operations on primary

**What is the expected behavior?**
_debug output:_
`Mongoose: models.distinct('field', {}, { readPreference: ReadPreference { mode: 'secondary', tags: undefined }})`
query does not appear in active operations on primary

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**
Node 12.13.1, Mongoose 5.9.9, MongoDB 4.2.5

**Comment by clhuang** [comment]
Proposed fix in #8906

**Comment by vkarpov15** [comment]
Fixed by #8906
